modules/table/src/integrationTest/java/org/apache/ignite/distributed/ItTxDistributedTestSingleNode.java

```diff
@@ -178,14 +178,13 @@ protected TxManager clientTxManager() {
 
     /** {@inheritDoc} */
     @Override
-    protected TxManager txManager(Table t) {
-        var clients = txTestCluster.raftClients.get(t.name());
+    protected TxManager txManager(TableImpl t) {
 
-        Peer leader = clients.get(0).leader();
+        CompletableFuture<ReplicaMeta> primaryReplica = txTestCluster.placementDriver.getPrimaryReplica(
+                new TablePartitionId(t.tableId(), 0),
+                txTestCluster.clocks.get(txTestCluster.localNodeName).now());
 
-        assertNotNull(leader);
-
-        TxManager manager = txTestCluster.txManagers.get(leader.consistentId());
+        TxManager manager = txTestCluster.txManagers.get(primaryReplica.join().getLeaseholder());
```

Comment:
I'd rather add willSucceedFast() or isDone() or any similar check before join. Currently it's not that important, however it might ease the debug in future if we will use another placement driver implementation instead of current test one and join will hang.